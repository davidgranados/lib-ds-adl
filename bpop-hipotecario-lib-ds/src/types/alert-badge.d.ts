import { ReactNode } from "react";
import { Testable } from "./global";

export interface AlertBadgeProps extends Testable {
  children?: ReactNode;
  type?: "info" | "warning" | "success" | "danger";
}
