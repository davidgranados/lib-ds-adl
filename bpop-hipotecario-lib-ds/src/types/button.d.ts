import { HTMLPropsFor } from "./component";

export type ButtonVariant = "primary" | "secondary";
export type ButtonTextSize = "sm" | "md" | "lg";
export type ButtonSize = "xs" | ButtonTextSize;

export interface DefaultButtonProps extends HTMLPropsFor<"button"> {
  label: string;
  variant?: ButtonVariant;
  disabled?: boolean;
  onClick?: () => void;
}

export interface ButtonProps extends DefaultButtonProps {
  size?: ButtonSize;
}

export interface ButtonTextProps extends DefaultButtonProps {
  size?: ButtonTextSize;
}
