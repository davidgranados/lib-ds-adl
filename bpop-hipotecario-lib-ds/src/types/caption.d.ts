import { ReactNode } from "react";
import { Testable } from "./global";

export declare type InputFieldCaptionStatus = "default" | "error" | "warning" | "success";

export declare interface InputFieldCaptionProps extends Testable {
  children: ReactNode;
  status?: InputFieldCaptionStatus;
}
