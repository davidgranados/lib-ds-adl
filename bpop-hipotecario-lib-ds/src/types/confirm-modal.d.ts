import { Testable } from "./global";

export declare interface ConfirmModalProps extends Testable {
  title: string;
  description?: string;
  show: boolean;
  mainButtonText?: string;
  secondaryButtonText?: string;
  onMainButtonClick?: () => void;
  onSecondaryButtonClick?: () => void;
  onToggle?: (state: boolean) => void;
}
