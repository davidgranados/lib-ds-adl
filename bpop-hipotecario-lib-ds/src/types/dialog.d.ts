import { Testable } from "./global";

export declare interface DialogProps extends Testable {
  show: boolean;
  width?: number;
  onToggle?: (state: boolean) => void;
}
