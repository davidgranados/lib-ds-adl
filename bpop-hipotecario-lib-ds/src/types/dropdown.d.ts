import { Testable } from "./global";

export declare type DropdownItems = { label: string; value: string | number }[];
export declare type DropdownOnChange = (v: string | number) => void;
export declare type DropdownValue = string | number;

export declare interface DropdownProps extends Testable {
  label: string;
  items: DropdownItems;
  disabled?: boolean;
  hasError?: boolean;
  caption?: string;
  placeholder?: string;
  value?: DropdownValue;
  onChange: DropdownOnChange;
}
