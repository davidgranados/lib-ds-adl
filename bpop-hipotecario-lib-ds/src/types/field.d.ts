import { ChangeEventHandler, FocusEventHandler, FormEventHandler } from "react";

import { InputBaseSize, InputBaseStatus } from "../../atoms/input-base";
import { TypographyStatus } from "../../atoms/typography";

export interface DefaultFieldProps {
  id: string;
  label: string;
  value?: string | number;
  status?: InputBaseStatus | TypographyStatus;
  caption?: string;
  placeholder?: string;
  size?: InputBaseSize;
  disabled?: boolean;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  onFocus?: FocusEventHandler<HTMLInputElement>;
  onBlur?: FocusEventHandler<HTMLInputElement>;
  onInput?: FormEventHandler<HTMLInputElement>;
}
