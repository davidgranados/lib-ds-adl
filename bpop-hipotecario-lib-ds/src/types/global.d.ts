import React from "react";

export type HTMLPropsFor<Tag extends keyof JSX.IntrinsicElements> = JSX.IntrinsicElements[Tag];
export type CombinedRef = Array<React.MutableRefObject<null> | React.ForwardedRef<HTMLInputElement>>;
