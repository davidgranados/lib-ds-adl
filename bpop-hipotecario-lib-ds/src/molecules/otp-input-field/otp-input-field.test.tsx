import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { OtpInputFieldProps } from "../../types";

import OtpInputField from "./otp-input-field";

describe("<OtpInputField />", () => {
  const onChange = jest.fn();
  beforeEach(() => {
    jest.clearAllMocks();
  });
  const getComponent = (componentProps?: OtpInputFieldProps) => <OtpInputField {...componentProps} />;
  it("should match snapshot", () => {
    const { container } = render(getComponent());
    expect(container).toMatchSnapshot();
  });
  it("should match hasError snapshot", () => {
    const { container } = render(getComponent({ hasError: true }));
    expect(container).toMatchSnapshot();
  });
  it("should accept initial value", () => {
    // given
    render(getComponent({ value: "123", numInputs: 3 }));
    const inputZero = screen.getByTestId("opt-input-field-item-0");
    const inputOne = screen.getByTestId("opt-input-field-item-1");
    const inputTwo = screen.getByTestId("opt-input-field-item-2");
    // when
    // then
    expect(inputZero).toHaveValue("1");
    expect(inputOne).toHaveValue("2");
    expect(inputTwo).toHaveValue("3");
  });
  it("should handle input change", async () => {
    // given
    render(getComponent({ onChange }));
    const inputZero = screen.getByTestId("opt-input-field-item-0");
    const inputOne = screen.getByTestId("opt-input-field-item-1");
    const inputValue = "1";
    // when
    userEvent.type(inputZero, inputValue);
    // then
    await waitFor(() => {
      expect(onChange).toHaveBeenCalledTimes(1);
      expect(inputZero).toHaveValue(inputValue);
      expect(inputOne).toHaveFocus();
    });
  });
  it("should handle backspace keydown", async () => {
    // given
    render(getComponent({ onChange }));
    const inputZero = screen.getByTestId("opt-input-field-item-0");
    const inputOne = screen.getByTestId("opt-input-field-item-1");
    const inputTwo = screen.getByTestId("opt-input-field-item-2");
    const inputValue = "1";
    // when
    userEvent.type(inputZero, inputValue);
    userEvent.type(inputOne, inputValue);
    userEvent.type(inputTwo, "{backspace}");
    userEvent.type(inputOne, "{backspace}");
    // then
    await waitFor(() => {
      expect(onChange).toHaveBeenCalledTimes(4);
      expect(inputZero).toHaveValue(inputValue);
      expect(inputOne).toHaveValue("");
      expect(inputZero).toHaveFocus();
    });
  });
  it("should handle delete keydown", async () => {
    // given
    render(getComponent({ onChange }));
    const inputZero = screen.getByTestId("opt-input-field-item-0");
    const inputOne = screen.getByTestId("opt-input-field-item-1");
    const inputValue = "1";
    // when
    userEvent.type(inputZero, inputValue);
    userEvent.type(inputOne, "{backspace}");
    userEvent.type(inputZero, "{del}");
    // then
    await waitFor(() => {
      expect(onChange).toHaveBeenCalledTimes(3);
      expect(inputZero).toHaveValue("");
      expect(inputOne).toHaveValue("");
      expect(inputOne).toHaveFocus();
    });
    // screen.debug();
  });
  it("should handle arrow right keydown", async () => {
    // given
    render(getComponent());
    const inputZero = screen.getByTestId("opt-input-field-item-0");
    const inputOne = screen.getByTestId("opt-input-field-item-1");
    // when
    userEvent.click(inputZero);
    userEvent.type(inputZero, "{arrowright}");
    // then
    await waitFor(() => {
      expect(inputOne).toHaveFocus();
    });
  });
  it("should handle arrow left keydown", async () => {
    // given
    render(getComponent());
    const inputZero = screen.getByTestId("opt-input-field-item-0");
    const inputOne = screen.getByTestId("opt-input-field-item-1");
    // when
    userEvent.click(inputZero);
    userEvent.type(inputZero, "{arrowright}");
    userEvent.type(inputOne, "{arrowleft}");
    // then
    await waitFor(() => {
      expect(inputZero).toHaveFocus();
    });
  });
  it("should prevent space keydown", async () => {
    // given
    render(getComponent());
    const inputZero = screen.getByTestId("opt-input-field-item-0");
    // when
    userEvent.click(inputZero);
    userEvent.type(inputZero, "{space}");
    // then
    await waitFor(() => {
      expect(inputZero).toHaveFocus();
    });
  });
  it("should prevent letter input", async () => {
    // given
    render(getComponent());
    const inputZero = screen.getByTestId("opt-input-field-item-0");
    // when
    userEvent.click(inputZero);
    userEvent.type(inputZero, "a");
    // then
    await waitFor(() => {
      expect(inputZero).toHaveValue("");
    });
  });
});
