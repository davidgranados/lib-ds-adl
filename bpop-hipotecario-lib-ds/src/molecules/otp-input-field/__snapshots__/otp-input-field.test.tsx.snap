// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OtpInputField /> should match hasError snapshot 1`] = `
<div>
  <div
    class="wrapper"
    data-testid="opt-input-field"
    id="opt-input-field"
  >
    <div
      class="inputs-wrapper error"
    >
      <div
        data-testid="opt-input-field-item-wrapper-0"
        id="opt-input-field-item-wrapper-0"
      >
        <input
          autocomplete="off"
          class="default"
          data-testid="opt-input-field-item-0"
          id="opt-input-field-item-0"
          maxlength="1"
          placeholder="-"
          type="tel"
          value=""
        />
      </div>
      <div
        data-testid="opt-input-field-item-wrapper-1"
        id="opt-input-field-item-wrapper-1"
      >
        <input
          autocomplete="off"
          class="default"
          data-testid="opt-input-field-item-1"
          id="opt-input-field-item-1"
          maxlength="1"
          placeholder="-"
          type="tel"
          value=""
        />
      </div>
      <div
        data-testid="opt-input-field-item-wrapper-2"
        id="opt-input-field-item-wrapper-2"
      >
        <input
          autocomplete="off"
          class="default"
          data-testid="opt-input-field-item-2"
          id="opt-input-field-item-2"
          maxlength="1"
          placeholder="-"
          type="tel"
          value=""
        />
      </div>
      <div
        data-testid="opt-input-field-item-wrapper-3"
        id="opt-input-field-item-wrapper-3"
      >
        <input
          autocomplete="off"
          class="default"
          data-testid="opt-input-field-item-3"
          id="opt-input-field-item-3"
          maxlength="1"
          placeholder="-"
          type="tel"
          value=""
        />
      </div>
      <div
        data-testid="opt-input-field-item-wrapper-4"
        id="opt-input-field-item-wrapper-4"
      >
        <input
          autocomplete="off"
          class="default"
          data-testid="opt-input-field-item-4"
          id="opt-input-field-item-4"
          maxlength="1"
          placeholder="-"
          type="tel"
          value=""
        />
      </div>
      <div
        data-testid="opt-input-field-item-wrapper-5"
        id="opt-input-field-item-wrapper-5"
      >
        <input
          autocomplete="off"
          class="default"
          data-testid="opt-input-field-item-5"
          id="opt-input-field-item-5"
          maxlength="1"
          placeholder="-"
          type="tel"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<OtpInputField /> should match snapshot 1`] = `
<div>
  <div
    class="wrapper"
    data-testid="opt-input-field"
    id="opt-input-field"
  >
    <div
      class="inputs-wrapper inputs-wrapper-enabled"
    >
      <div
        data-testid="opt-input-field-item-wrapper-0"
        id="opt-input-field-item-wrapper-0"
      >
        <input
          autocomplete="off"
          class="default"
          data-testid="opt-input-field-item-0"
          id="opt-input-field-item-0"
          maxlength="1"
          placeholder="-"
          type="tel"
          value=""
        />
      </div>
      <div
        data-testid="opt-input-field-item-wrapper-1"
        id="opt-input-field-item-wrapper-1"
      >
        <input
          autocomplete="off"
          class="default"
          data-testid="opt-input-field-item-1"
          id="opt-input-field-item-1"
          maxlength="1"
          placeholder="-"
          type="tel"
          value=""
        />
      </div>
      <div
        data-testid="opt-input-field-item-wrapper-2"
        id="opt-input-field-item-wrapper-2"
      >
        <input
          autocomplete="off"
          class="default"
          data-testid="opt-input-field-item-2"
          id="opt-input-field-item-2"
          maxlength="1"
          placeholder="-"
          type="tel"
          value=""
        />
      </div>
      <div
        data-testid="opt-input-field-item-wrapper-3"
        id="opt-input-field-item-wrapper-3"
      >
        <input
          autocomplete="off"
          class="default"
          data-testid="opt-input-field-item-3"
          id="opt-input-field-item-3"
          maxlength="1"
          placeholder="-"
          type="tel"
          value=""
        />
      </div>
      <div
        data-testid="opt-input-field-item-wrapper-4"
        id="opt-input-field-item-wrapper-4"
      >
        <input
          autocomplete="off"
          class="default"
          data-testid="opt-input-field-item-4"
          id="opt-input-field-item-4"
          maxlength="1"
          placeholder="-"
          type="tel"
          value=""
        />
      </div>
      <div
        data-testid="opt-input-field-item-wrapper-5"
        id="opt-input-field-item-wrapper-5"
      >
        <input
          autocomplete="off"
          class="default"
          data-testid="opt-input-field-item-5"
          id="opt-input-field-item-5"
          maxlength="1"
          placeholder="-"
          type="tel"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
